h5netcdf/core.py

```diff
@@ -622,6 +695,23 @@ def __init__(self, path, mode='a', invalid_netcdf=None, **kwargs):
         # if we actually use invalid NetCDF features.
         self._write_ncproperties = invalid_netcdf is not True
 
+        # phony dimension handling
+        if phony_dims is not None:
+            if phony_dims == 'sort':
+                self._phony_dim_count = 0
+                self._labeled_dim_count = 0
+            elif phony_dims == 'access':
+                self._phony_dim_count = 0
+                self._labeled_dim_count = None
```

Comment:
Encoding the phony dimension mode in the type of these arguments feels a little too clever/indirect to me. It is not obvious that `labeled_dim_count is None` means "per access" mode.

Can you store the phony dimension mode explicitly in an attribute, e.g., `self._phony_dims_mode = phony_dims`? I think that would make the other code a little easier to read.

Reply:
Yes, this makes absolutely sense. I'll add this.